### ISSUE TITLE
better message for blocked users

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#8636]
Currently, when a moderator blocks a user, the user sees this message if they are already logged in or are trying to log in:

![image](https://user-images.githubusercontent.com/1495809/33219829-825bd7e4-d13c-11e7-93d6-15c27de255b2.png)

We want the message to be replaced with this text:

**"This account, Profile Name {insert profile name here}, User ID {insert User ID here}, has been blocked for breaking the [Community Guidelines](https://habitica.com/static/community-guidelines) or [Terms of Service](https://habitica.com/static/terms). For details or to ask to be unblocked, please email Lemoness at leslie@habitica.com or ask your parent or guardian to email her. Please copy your Profile Name and User ID into the email."**

The links for the Community Guidelines and Terms of Service are https://habitica.com/static/community-guidelines and https://habitica.com/static/terms

The email address should be a mailto link and must use the environment variable `EMAILS.COMMUNITY_MANAGER_EMAIL` instead of having "leslie@habitica.com" hard-coded. To see examples of how to do that, use `git grep COMMUNITY_MANAGER_EMAIL`.

We also want the format of the messages to be changed.  Instead of seeing the red notification shown above, they should see a modal that displays the message. The links must be clickable and the text must be selectable for copying. The modal does not need a button for closing it since the website can't be used when you're blocked, but if you want to include a button (or if anyone else here asks that one be included), that's fine.

If it's difficult to stop the red notification from appearing, then it's okay if they see that as well as the modal.

For testing, you can set a user account to be blocked by editing their document in the database to set `auth.blocked` to `true`, or set one account to be an admin user (using the Debug menu in the footer of your test website), and then go to Social > Hall to block a different account. The interface is shown below. Enter the User ID, click submit, then when the details open up, click the "Auth" header. 

![image](https://cloud.githubusercontent.com/assets/1495809/24584149/d813e5aa-17a6-11e7-875a-f109f40a829b.png)

**Comment by rhdsmnd** [comment]
Is there a way to get the Community Manager profile name ("Lemoness")?  In the config.json, there seems to only be an entry for COMMUNITY_MANAGER_EMAIL and not other user details.  Peeking in my own (sparse) test database, there doesn't seem to be any references to a Community Manager (or other special staff/admin users) either.

Also, are there other versions of the string ("This account...") for the other locales?

**Comment by Alys** [comment]
@rhdsmnd There isn't an environment variable for "Lemoness" and we're not too concerned about that. Our main reason for having the env variables is so that email addresses can be changed quickly when Lemoness is unavailable. It's okay if the messages still show her name then (slightly inaccurate but it doesn't stop people from getting help quickly).

To find the strings that contain the current message you can do commands like this: `git grep "Account has been"`. You can edit the string(s) in the `en` locale.

You never need to worry about any locales other than `en`, and in fact you should specifically not edit the other locales. When the `en` locales files are changed, we upload them to Transifex and then our translators modify the other locales.

**Comment by kvsm** [comment]
Is this in progress? If not, I'd be happy to take a look.

**Comment by Alys** [comment]
@rhdsmnd Are you working on this? If so, that's great! If not, @kvsm can. :)

**Comment by rhdsmnd** [comment]
Yeah, I've been tinkering with it and will start coding the new message;
sorry about going mia.  I ran into weird behavior when setting auth.blocked
(logged in user still able to use webite?).

**Comment by Alys** [comment]
@rhdsmnd No worries and thank you!

The logged in user should still see the website and can probably navigate around but they can't take any actions such as completing or editing tasks, buying items (gems or gold), posting messages. Once they've logged out, they can't log back in.  Does that explain what you're seeing?

**Comment by FrankieFabuloso** [comment]
Is this issue still open? I'd be happy to work on it. 😄

**Comment by Alys** [comment]
@FrankieFabuloso Ah, I'm sorry, we should have had this marked as on hold. It will require changes to the website, and a redesigned website will be released soon. We'll take this off hold after that. Sorry about that! If you'd like to work on it then, please post again here when the new site is in use.

**Comment by FrankieFabuloso** [comment]
@Alys No worries! I'll look for another to work on!

**Comment by librarianmage** [comment]
Should this still be on hold? How does it look with the new update?

**Comment by Alys** [comment]
@MathWhiz Thanks for asking! I've updated the top post to make it relevant for the new website, and I've changed the labels. The requirements are still much the same.

Are any of the contributors who've commented here interested in working on this now?

**Comment by TheHollidayInn** [comment]
I'm not sure we will be able to include the profile name because the user route is not authorized when the user is banned.

**Comment by Alys** [comment]
No worries. It's fine to leave out the name for that reason.
